s3/src/main/scala/geotrellis/store/s3/util/S3RangeReader.scala

```diff
@@ -39,19 +39,14 @@ class S3RangeReader(
   client: S3Client
 ) extends RangeReader {
 
-  lazy val metadata: HeadObjectResponse = {
-    val headRequest =
-      HeadObjectRequest
-        .builder()
-        .bucket(request.bucket)
-        .key(request.key)
-        .build()
+  lazy val totalLength: Long = {
+    val responseStream = client.getObject(request)
+    val length = responseStream.response.contentLength
```

Comment:
Since we can just use a `GET` query to produce our `GetObjectResponse`, we don't need to worry about whether or not to reader the `HEADER` of the object.

Reply:
@jbouffard I though [here](https://github.com/locationtech/geotrellis/issues/2889#issuecomment-511799089) was mentioned that we want to replicate [HttpRangeReader](https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/store/http/util/HttpRangeReader.scala#L34) approach.

Reply:
@pomadchin Is there a problem with using `Get` for everything? I'm not really familiar with that operation.

Reply:
We would like to use HEAD if possible (I think). 

See the description of [HEAD](https://docs.aws.amazon.com/en_us/AmazonS3/latest/API/RESTObjectHEAD.html) vs [GET](https://docs.aws.amazon.com/en_us/AmazonS3/latest/API/RESTObjectGET.html)

Reply:
I see. You're right that it looks like `HEAD` is preferable.